torch/csrc/utils/tensor_new.cpp

```diff
@@ -613,6 +613,23 @@ Tensor sparse_csr_tensor_ctor(c10::DispatchKey dispatch_key, at::ScalarType scal
   const int NUM_ARGS = 9, CROW_INDICES_ARG = 0, COL_INDICES_ARG = 1, VALUES_ARG = 2;
   ParsedArgs<NUM_ARGS> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
+  auto safe_get_attr_string = [](PyObject *o, const char *attr_name) -> PyObject* {
+    // Clear error indicator if attribute does not exists.
+    // Otherwise subsequent Python C API calls might return bogus values.
+    // See https://github.com/pytorch/pytorch/issues/58520 for more details
+    auto rc = PyObject_GetAttrString(o, attr_name);
+    if (!rc) {
```

Comment:
Notice that you have to check if the exception is really an AttributeError, at least. If it is not, the exception should be raised.
For instance, say, `o` is an instance of the following class:
```
class A:
    @property
    def dtype(self):
        raise NotImplementedError("Say something")
```
in which case the error should not be cleared.

So, use the following logic:
```
if (!rc) {
    if (!PyErr_ExceptionMatches(PyExc_AttributeError)) {
        throw python_error();
    }
    // Warning: a wrong attribute error may be suppressed here
    PyErr_Clear();
}
```

Reply:
Ignore the above comment if it is not a part of this PR. Nevertheless, the current behaviour must be fixed at some point.

Reply:
Resolving via https://github.com/pytorch/pytorch/issues/58735

Reply:
Thanks Pearu, I will add this logic too.